### PR TITLE
⌚️ Add format_time filter tag that consider the org timezone and date fo…

### PR DESCRIPTION
**Pull request by norkans7**
…rmat set

### CODE REVIEW

File: temba/utils/templatetags/temba.py [hunk 1]
```diff
@@ -168,3 +173,29 @@ def to_json(value):
...
+    now = timezone.now()
+    twelve_hours_ago = now - timedelta(hours=12)
+
+    if dtime > twelve_hours_ago:
+        return "%d:%s %s" % (int(dtime.strftime("%I")), dtime.strftime("%M"), dtime.strftime("%p").lower())
+    elif now.year == dtime.year:
+        return "%s %d" % (dtime.strftime("%b"), int(dtime.strftime("%d")))
+    else:
+        if month_first:
+            return "%d/%d/%s" % (int(dtime.strftime("%m")), int(dtime.strftime("%d")), dtime.strftime("%y"))
```
Comment (rowanseymour) [inline temba/utils/templatetags/temba.py]: is there an argument for this matching the format we use elsewhere, i.e. 

captura de pantalla 2019-01-10 a la s 10 11 09

Comment (norkans7) [reply temba/utils/templatetags/temba.py]: Day first as `12 Oct 2018`
Month first as `Oct 12, 2018` ?

Comment (rowanseymour) [reply temba/utils/templatetags/temba.py]: cc @ericnewcomer @nicpottier

Comment (norkans7) [reply temba/utils/templatetags/temba.py]: Also we could be consistent with this filter everywhere we use a datetime field

Comment (nicpottier) [reply temba/utils/templatetags/temba.py]: I think we are losing the forest from the trees a bit here. day first / month first really only make sense as settings when we are talking purely numeric dates no? It doesn't seem like that really applies once you start using month names does it? Is Europe different than the US in that respect? Does anyone say "5 January, 2013" vs "January 5, 2013"? Seems like really we should be doing "January 5th, 2013" if we are going through the trouble of a specific tag for the more verbose version and use that in both cases. But I don't think that actually applies in this case, `gmail_time` is meant to be used in lists and present something pretty terse, it shouldn't be how we show dates everywhere. Perhaps that is a hint as to what we should call it (`short_time`?).

So I think the only thing that needs to change with respect to orgs is the most terse version that is purely numeric should be in the org format. I think this can be simplified further by just looking for `org` in the context in our template tag and using the date time format in there if present so we don't have to pass it in everywhere.

So in summary:
 1. I think the tag should be called something like `short_time` and take just a date
 2. It should infer month / day first by looking in the context for an `org` var and using that setting, otherwise defaulting to day first
 3. nothing else needs to change on it, it is fine for the short version to be "Jan 5" when we aren't including year

Extra credit:
 1. We could have a nicer date formatting tag that adds "nd" "th" to dates and use that everywhere else for more verbose times. `pretty_time`? ("October 13th, 2018")

Thoughts?

Comment (rowanseymour) [reply temba/utils/templatetags/temba.py]: I'm fine with this being `short_time` only used on lists, but I do think all date formatting should follow the org setting. It's certainly what that user expected when he changed the setting. People in the rest of the world do actually write "1 Jan 2018"

FYI this is how my gmail appears:

captura de pantalla 2019-01-11 a la s 12 02 00

And this is from the BBC homepage:

captura de pantalla 2019-01-11 a la s 12 05 33

Comment (ericnewcomer) [reply temba/utils/templatetags/temba.py]: I personally don't like ordinal dates especially when there is a year involved. But ya, it would seem day first only matters in the case of numeric date formats like M/D/Y. 

So another proposal would be:

Large:
January 11, 2019

Medium:
Jan 11, 2019

Small (if we want ordinals, then i don't hate them here)
Jan 11th (or if year is important then, 1/11/19)

Comment (nicpottier) [reply temba/utils/templatetags/temba.py]: Just so we are on the same page, what do you mean by ordinal date? Wikipedia says that is day of the year (1-365).

Jesus, `1 Jan` on gmail? What is wrong with the world? I think the person complained not about that case for what its worth but the full numeric case, ie `10/10/2018`. (may be wrong, but I think so) But ok, ya I guess we change the ordering on the medium form as well in `short_date`

I really don't want three different tags. Can we just have two? `short_date` and `pretty_date`?

`short_date` is:

`3:44pm` or `15:44` if on the same day
`Jan 10` or `10 Jan` if in the same year
`6/24/2018` or `24/6/2018` if in different year

That would be most consistent with what we have now.

`pretty_date` is always:

"January 10th, 2018" or "10th January, 2018"

Comment (ericnewcomer) [reply temba/utils/templatetags/temba.py]: Ooof, they really write 11 January? Just did some poking around and I guess that is really a thing.  Looks like same deal for ordinals, I see lots of 11th January too. Seems we should honor org date order in all forms.

Comment (rowanseymour) [reply temba/utils/templatetags/temba.py]: @nicpottier only tweak would be you don't put a comma in `10th January 2018`.. the comma in `January 10th, 2018` is a subtle recognition that the ordering makes no sense 😛

Comment (ericnewcomer) [reply temba/utils/templatetags/temba.py]: Forgot to mention, my vote is pretty much what @nicpottier said with two date levels. My only bias would be to drop the ordinals on the pretty dates but willing to cede to the majority there.

Comment (nicpottier) [reply temba/utils/templatetags/temba.py]: Just occurred to me that maybe the short version should be called `short_datetime` since ya it includes a time. We can eventually ad a `pretty_datetime` when we want both date and time displayed.

File: templates/contacts/contact_history.haml [hunk 2]
```diff
@@ -100,9 +100,9 @@
     %td.created_on
       %span.time
         %span.short
-          {{item.time|gmail_time}}
+          {% short_datetime item.time %}
         %span.long.hide
-          {{item.time}}
+          {% pretty_datetime item.time %}
```
Comment (norkans7) [inline templates/contacts/contact_history.haml]: `pretty_datetime` is needed here
